arangod/Aql/RegisterPlan.cpp

```diff
@@ -144,6 +172,15 @@ void RegisterPlanWalkerT<T>::after(T* en) {
     planRegistersForCurrentNode(en);
   }
 
+  updateRegsToKeep(en, en->getVarsUsedLater(), en->getVarsValid());
+
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  auto actual = plan->calcRegsToKeep(en->getVarsUsedLaterStack(), en->getVarsValidStack(),
+                                     en->getVariablesSetHere());
+  TRI_ASSERT(regsToKeepStack == actual);
+#endif
+
+  en->setRegsToKeep(regsToKeepStack);
```

Comment:
is it better to move here?
```suggestion
  en->setRegsToKeep(std::move(regsToKeepStack));
```

Reply:
We need a copy for every node. Maybe we can improve that later with `immer`.

Reply:
Due to the variable naming, I was assuming that `regsToKeepStack`is a local variable.
But it seems to be an instance variable, and if so, I do understand.

Reply:
No. This is an intentional copy. `regsToKeepStack` is updated for every node. (see above `regsToKeepStack.back().clear()`) Each node received a copy of the stack. To reduce those copies we have planned to replace this by an immutable data structure.